src/orion/core/utils/__init__.py

```diff
@@ -72,7 +77,14 @@ def _import_modules(cls):
     # base = import_module(cls.__base__.__module__)
 
     # Get types advertised through entry points!
-    for entry_point in pkg_resources.iter_entry_points(cls.__name__):
+    entry_point_iterable = (
+        entry_points()
+        if sys.version_info >= (3, 10)
+        else iter_entry_points(cls.__name__)
+    )
+    for entry_point in entry_point_iterable:
+        if sys.version_info >= (3, 10) and entry_point.group != cls.__name__:
```

Comment:
IMHO it would be cleaner if that if statement could be contained above the for loop, maybe as a list comprehension in the if-else of the `entry_point_iterable` assignement. So later on when 3.10 is the oldest version supported we can clear out the if-else and leave the rest as-is.

Reply:
Yes, good thinking.  Will do that.

Reply:
Is this what you had in mind?

Reply:
Is this what you had in mind?